src/test_l1_dsHdmiIn.c

```diff
@@ -456,26 +491,56 @@ void test_l1_dsHdmiIn_negative_dsHdmiInGetStatus(void) {
  * |02|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_0, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 0 should be selected |
  * |03|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_1, bool: true, dsVideoPlane_SECONDARY, bool true| dsERR_NONE | Port 1 should be selected |
  * |04|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_2, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 2 should be selected |
- * |05|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
+ * |05|Verify that the dsHdmiInSelectPort function successfully sets up the port for sink and source devices 
+ * |06|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
  * 
  */
 void test_l1_dsHdmiIn_positive_dsHdmiInSelectPort(void) {
     gTestID = 9;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID); 
-    // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
-    UT_ASSERT_EQUAL(dsHdmiInInit(), dsERR_NONE);
+    dsError_t result;
+    uint8_t numInputs = 0;
 
-    // Step 2: Call dsHdmiInSelectPort() to select Port 0
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_0, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
-
-    // Step 3: Call dsHdmiInSelectPort() to select Port 1
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_1, true, dsVideoPlane_SECONDARY, true), dsERR_NONE);
-
-    // Step 4: Call dsHdmiInSelectPort() to select Port 2
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_2, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
+    // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
+    UT_ASSERT_EQUAL_FATAL(dsHdmiInInit(), dsERR_NONE);
+
+    // Step 2: Loop through all combinations of dsHdmiInPort_t and dsVideoPlaneType_t
+    for (uint8_t i = dsHDMI_IN_PORT_0; i < numInputs; i++)
+    {
+        for(int plane = dsVideoPlane_PRIMARY; plane < dsVideoPlane_MAX; plane++)
+        {
+          UT_LOG_DEBUG("Invoking dsHdmiInSelectPort() with port number %d", i);
+          result = dsHdmiInSelectPort(i, false, plane, false);
```

Comment:
You can use two loops: one from dsHDMI_IN_PORT_0 to numInputs and another from numInputs to MAX.

 Inside these loops, use the dsVideoPlane_PRIMARY and dsVideoPlane_SECONDARY values separately, rather than nesting them.